src/Admin/Editor.php

```diff
@@ -35,6 +36,19 @@ public function admin_init() {
 		add_filter( 'wp_editor_settings', [ $this, 'wp_editor_settings' ], 10, 2 );
 	}
 
+	/**
+	 * Fires after a post is restored from the Trash.
+	 *
+	 * @param int    $post_id         Post ID.
+	 * @param string $previous_status The status of the post at the point where it was trashed.
+	 * @return void
+	 */
+	public function untrashed_post_cb( $post_id, $previous_status ) {
+		// If have errors when validating the post content/data, do not show those in the admin when untrash.
```

Comment:
Something to this tune, perhaps:

```suggestion
		// If have errors when validating the post content/data, do not show those in the admin when untrash.
		$untrashed_post = get_post( $post_id );
		
		// Bail if the untrashed post is not a GraphQL Document
		if ( ! isset( $untrashed_post->post_type ) || Document::TYPE_NAME !== $untrashed_post->post_type ) {
		   return;
		}
	
```